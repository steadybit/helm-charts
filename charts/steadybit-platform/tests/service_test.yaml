templates:
  - services.yaml
chart:
  appVersion: 0.0.0
  version: 0.0.0
tests:
  - it: should use deprecated service type
    set:
      platform:
        service:
          type: NodePort
    asserts:
      - matchSnapshot: { }
  - it: should use service type
    set:
      service:
          type: ClusterIP
    asserts:
      - matchSnapshot: { }
  - it: should use annotations
    set:
      service:
        annotations:
          this: that
    asserts:
      - matchSnapshot: { }