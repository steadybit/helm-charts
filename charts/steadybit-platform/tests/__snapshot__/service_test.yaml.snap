should use annotations:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        this: that
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      ports:
        - name: ui-port
          port: 80
          protocol: TCP
          targetPort: 8080
        - name: websocket-port
          port: 7878
          protocol: TCP
          targetPort: 7878
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: steadybit-platform
      type: null
should use deprecated service type:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      ports:
        - name: ui-port
          port: 80
          protocol: TCP
          targetPort: 8080
        - name: websocket-port
          port: 7878
          protocol: TCP
          targetPort: 7878
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: steadybit-platform
      type: NodePort
should use service type:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: steadybit-platform
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: steadybit-platform-0.0.0
      name: RELEASE-NAME-steadybit-platform
      namespace: NAMESPACE
    spec:
      ports:
        - name: ui-port
          port: 80
          protocol: TCP
          targetPort: 8080
        - name: websocket-port
          port: 7878
          protocol: TCP
          targetPort: 7878
      selector:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: steadybit-platform
      type: ClusterIP
